crates/solver/src/solver/http_solver.rs

```diff
@@ -103,14 +103,30 @@ impl HttpSolver {
         &self,
         auction_id: AuctionId,
         run_id: u64,
-        orders: Vec<LimitOrder>,
+        mut orders: Vec<LimitOrder>,
         liquidity: Vec<Liquidity>,
         gas_price: f64,
         external_prices: ExternalPrices,
     ) -> Result<(BatchAuctionModel, SettlementContext)> {
+        // The HTTP solver interface expects liquidity limit orders (like 0x
+        // limit orders) to be added to the `orders` models and NOT the
+        // `liquidity` models. Split the two here to avoid indexing errors
+        // later on.
+        let (limit_orders, amms) = liquidity.into_iter().fold(
+            (Vec::new(), Vec::new()),
+            |(mut limit_orders, mut amms), liquidity| {
+                match liquidity {
+                    Liquidity::LimitOrder(limit_order) => limit_orders.push(limit_order),
+                    amm => amms.push(amm),
+                }
+                (limit_orders, amms)
+            },
+        );
```

Comment:
nit: I think this could probably look a little cleaner by using `Iterator::partition()`.

Reply:
Also wanted to use `partition`, but you can't because each partition has a different type.

However, I found [`Itertools::partition_map`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.partition_map) and will refactor to use it instead.

Thanks for the tip 😄.